ast.hh

```diff
@@ -555,10 +557,10 @@ public:
 			{
 				ASTStack *st = reinterpret_cast<ASTStack *>(d);
 				T *obj = new T();
-				InputRange r(b,e);
+				InputRange input(b,e);
```

Comment:
Could be range, that'd be fine. Not `r` because there's already an `r` in scope. `-Wshadow` warns about this, and I think it's a sensible thing to warn about: it's easy to get confused between `r` and `r` (or, as we had in another place, `i` and `i`).